From 4a5f86560f018a20ee36de45ef712559e1c66450 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Mon, 13 Jul 2020 22:01:27 -0400
Subject: mingw: Backport SVN 159364

config.gcc (i[34567]86-*-mingw* | x86_64-*-mingw*): Choose t-mingw-w64 or t-mingw-w32 for multilib configuration.

2010-05-13  Kai Tietz  <kai.tietz@onevision.com>

        * config.gcc (i[34567]86-*-mingw* | x86_64-*-mingw*): Choose
        t-mingw-w64 or t-mingw-w32 for multilib configuration.
        * config/i386/t-mingw-w32: New.
        * config/i386/t-mingw-w64 (MULTILIB_OSDIRNAMES): Change lib64 to lib.

From-SVN: r159364

Author:    Tim Stahlhut <stahta01@gmail.com>
---
 gcc/config.gcc              |  5 ++++-
 gcc/config/i386/t-mingw-w32 | 12 ++++++++++++
 gcc/config/i386/t-mingw-w64 |  2 +-
 3 files changed, 17 insertions(+), 2 deletions(-)
 create mode 100644 gcc/config/i386/t-mingw-w32

diff --git a/gcc/config.gcc b/gcc/config.gcc
index aa55f779946..eb2f786487c 100644
--- a/gcc/config.gcc
+++ b/gcc/config.gcc
@@ -1400,9 +1400,12 @@ i[34567]86-*-mingw* | x86_64-*-mingw*)
 	fi
 	tmake_file="${tmake_file} ${tmake_eh_file} i386/t-cygming"
         case ${target} in
-               *-w64-*)
+               x86_64-w64-*)
                		tmake_file="${tmake_file} i386/t-mingw-w64"
 			;;
+	       i[34567]86-w64-*)
+			tmake_file="${tmake_file} i386/t-mingw-w32"
+			;;
                *)
                		tmake_file="${tmake_file} i386/t-mingw32"
                      	;;
diff --git a/gcc/config/i386/t-mingw-w32 b/gcc/config/i386/t-mingw-w32
new file mode 100644
index 00000000000..62d6330fabb
--- /dev/null
+++ b/gcc/config/i386/t-mingw-w32
@@ -0,0 +1,12 @@
+# Match SYSTEM_INCLUDE_DIR
+NATIVE_SYSTEM_HEADER_DIR = /mingw/include
+
+MULTILIB_OPTIONS = m64/m32
+MULTILIB_DIRNAMES = 64 32
+MULTILIB_OSDIRNAMES = ../lib64 ../lib
+
+# MinGW-specific parts of LIB_SPEC
+SHLIB_LC = -lmingw32 -lmingwex -lmoldname -lmsvcrt -luser32 -lkernel32 -ladvapi32 -lshell32
+
+LIBGCC = stmp-multilib
+INSTALL_LIBGCC = install-multilib
diff --git a/gcc/config/i386/t-mingw-w64 b/gcc/config/i386/t-mingw-w64
index dbbe00a1d96..b0106e5e64b 100644
--- a/gcc/config/i386/t-mingw-w64
+++ b/gcc/config/i386/t-mingw-w64
@@ -3,7 +3,7 @@ NATIVE_SYSTEM_HEADER_DIR = /mingw/include
 
 MULTILIB_OPTIONS = m64/m32
 MULTILIB_DIRNAMES = 64 32
-MULTILIB_OSDIRNAMES = ../lib64 ../lib32
+MULTILIB_OSDIRNAMES = ../lib ../lib32
 
 # MinGW-specific parts of LIB_SPEC
 SHLIB_LC = -lmingw32 -lmingwex -lmoldname -lmsvcrt -luser32 -lkernel32 -ladvapi32 -lshell32
-- 
